File: gems/decomplex/examples/c/implicit-control-flow.c
typedef struct FlowExample { int status; int valid; int done; } FlowExample;
void prepare(FlowExample *self) { self->status = 1; }
void validate(FlowExample *self) { self->valid = self->status == 1; }
void commit(FlowExample *self) { self->done = self->valid; }
void ok1(FlowExample *self) { prepare(self); validate(self); commit(self); }
void ok2(FlowExample *self) { prepare(self); validate(self); commit(self); }
void ok3(FlowExample *self) { prepare(self); validate(self); commit(self); }
void ok4(FlowExample *self) { prepare(self); validate(self); commit(self); }
void drift(FlowExample *self) { validate(self); prepare(self); commit(self); }

File: gems/decomplex/examples/go/implicit-control-flow.go
package main
type FlowExample struct { status int; valid bool; done bool }
func (self *FlowExample) prepare() { self.status = 1 }
func (self *FlowExample) validate() { self.valid = self.status == 1 }
func (self *FlowExample) commit() { self.done = self.valid }
func (self *FlowExample) ok1() { self.prepare(); self.validate(); self.commit() }
func (self *FlowExample) ok2() { self.prepare(); self.validate(); self.commit() }
func (self *FlowExample) ok3() { self.prepare(); self.validate(); self.commit() }
func (self *FlowExample) ok4() { self.prepare(); self.validate(); self.commit() }
func (self *FlowExample) drift() { self.validate(); self.prepare(); self.commit() }
